internal/apps/dop/endpoints/pipeline.go

```diff
@@ -394,26 +394,11 @@ func (e *Endpoints) pipelineRun(ctx context.Context, r *http.Request, vars map[s
 		ConfigManageNamespaces: []string{utils.MakeUserOrgPipelineCmsNs(identityInfo.UserID, p.OrgID)},
 		Secrets:                utils.GetGittarSecrets(p.ClusterName, p.Branch, p.CommitDetail),
 	}); err != nil {
-		var apiError, ok = err.(*errorresp.APIError)
-		if !ok {
-			// Failed to convert to apiError type, return the error passed by the pipeline component
-			return errorresp.ErrResp(err)
-		}
-
-		ctxMap, ok := apiError.Ctx().(map[string]interface{})
-		if !ok {
-			// Interface converted to map[string]interface{} fails and returns the error passed by the pipeline component
-			return errorresp.ErrResp(err)
+		runningPipelineLink, ok := e.ProjectPipelineSvc.TryGenRunningPipelineLinkFromErr(p.PipelineDTO.OrgName, p.PipelineDTO.ProjectID, p.PipelineDTO.ApplicationID, err)
```

Comment:
You write the same logic in endpoints and services.
It's better to handle it only in one place, mostly in services.

Reply:
adjusted, TryGenRunningPipelineLinkFromErr return `err` and `bool`